src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,692 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents a repository rule.
+ */
+public class GHRepositoryRule {
+    private Type type;
+    private RulesetSourceType ruleset_source_type;
+    private String ruleset_source;
+    private long ruleset_id;
+    private Map<String, JsonNode> parameters;
```

Comment:
Ouch. I kind of hate the Response schema under https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-rules-for-a-branch . 

For Java, I prefer the API we expose for 
https://github.com/gheine/github-api/blob/main/src/main/java/org/kohsuke/github/GHBranchProtection.java but I'm not sure if it possible to translate. 

What we want is a way for people to easily walk the list of rules in a meaningful way.

Reply:
I don't have a better solution for this at this time.  Until someone comes up with a better solution, I'm comfortable with taking what we have.